packages/mdx/lib/plugin/recma-jsx-rewrite.js

```diff
@@ -356,13 +359,23 @@ export function recmaJsxRewrite(options = {}) {
             }
 
             for (const [
-              invalidComponentName,
-              variable
-            ] of invalidComponentNameToVariable) {
+              _componentsId,
+              _componentName
+            ] of idToInvalidComponentName) {
+              // For component IDs that render invalid JSX (ex. <_components.custom-element>),
+              // Generate a separate variable to index into `_components`
+              // i.e. `const _component0 = _components['custom-element']`
+              //      `return <_component0>...`
```

Comment:
```suggestion
              // For JSX IDs that can’t be represented as JavaScript IDs (as in,
              // those with dashes, such as `custom-element`), generate a
              // separate variable that is a valid JS ID (such as `_component0`),
              // and takes it from components:
              // `const _component0 = _components['custom-element']`
```

Taking your phrasing, I feel this suggestion is a bit more correct in how it is worded.
It also replaces Latin abbreviations with English words, which I believe makes them better readable to more people